clients/cfrestclient/rest_cloud_foundry_client_extended.go

```diff
@@ -68,23 +157,136 @@ func (c CloudFoundryRestClient) getSharedDomainsInternal(cloudFoundryUrlElements
 }
 
 func getPathQueryElements(response *models.CloudFoundryResponse) (CloudFoundryUrlElements, error) {
-	nextUrl, err := url.Parse(response.NextURL)
+	nextUrl, err := url.Parse(response.Pagination.Next.Href)
 	if err != nil {
-		return CloudFoundryUrlElements{}, fmt.Errorf("Could not parse next_url for getting shared domains: %s", response.NextURL)
+		return CloudFoundryUrlElements{}, fmt.Errorf("could not parse pagintaion.next for getting shared domains: %s", response.Pagination.Next.Href)
 	}
 	nextUrlQuery := nextUrl.Query()
 	page := nextUrlQuery.Get("page")
-	resultsPerPage := nextUrlQuery.Get("results-per-page")
-	orderDirection := nextUrlQuery.Get("order-direction")
+	resultsPerPage := nextUrlQuery.Get("per_page")
 
-	return CloudFoundryUrlElements{Page: &page, ResultsPerPage: &resultsPerPage, OrderDirection: &orderDirection}, nil
+	return CloudFoundryUrlElements{Page: &page, ResultsPerPage: &resultsPerPage}, nil
 
 }
 
-func toSharedDomains(response *models.CloudFoundryResponse) []models.SharedDomain {
-	var result []models.SharedDomain
+func toSharedDomains(response *models.CloudFoundryResponse) []models.Domain {
+	var result []models.Domain
 	for _, cloudResource := range response.Resources {
-		result = append(result, models.NewSharedDomain(cloudResource.Entity.Name, cloudResource.Metadata.GUID, cloudResource.Metadata.URL))
+		result = append(result, models.NewDomain(cloudResource.Name, cloudResource.GUID))
 	}
 	return result
 }
+
+func getPaginatedResources[T any](url, token string) ([]T, error) {
+	var result []T
+	for url != "" {
+		body, err := sendRequestAndReadBody(url, token)
+		if err != nil {
+			return nil, err
+		}
+		response, err := parseBody[models.PaginatedResponse[T]](body)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, entity := range response.Resources {
+			result = append(result, entity)
+		}
+		url = response.Pagination.NextPage
+	}
+	return result, nil
+}
+
+func getPaginatedResourcesWithIncluded[T any, Auxiliary any](url, token string, auxiliaryContentHandler func(T, Auxiliary) T) ([]T, error) {
+	var result []T
+	for url != "" {
+		body, err := sendRequestAndReadBody(url, token)
+		if err != nil {
+			return nil, err
+		}
+		response, err := parseBody[models.PaginatedResponseWithIncluded[T, Auxiliary]](body)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, entity := range response.Resources {
+			result = append(result, auxiliaryContentHandler(entity, response.Included))
+		}
+		url = response.Pagination.NextPage
+	}
+	return result, nil
+}
+
+func sendRequestAndReadBody(url, token string) ([]byte, error) {
```

Comment:
maybe rename to executeRequest, usually, a method should do one thing, but this can be considered as one operation